### PR TITLE
Migrate to jHipster-6.5.1 with Java_11

### PR DESCRIPTION
**Pull request by saylenty**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -142,223 +159,209 @@ task cucumberTestReport(type: TestReport) {
...
-
-checkstyle {
-    toolVersion = "7.6.1"
-    ignoreFailures = true
-    configFile = rootProject.file('./qa/checkstyle.xml')
-}
+    // XM custom dependencies
+    implementation "org.hibernate:hibernate-envers"
+    implementation 'com.vladmihalcea:hibernate-types-52:2.4.3'
+    implementation 'com.vladmihalcea:hibernate-types-52:2.4.3'
```
Comment (v-kyrychenko) [inline build.gradle]: why this dependencie duplicated? And what purpose of this library?

File: build.gradle [hunk 2]
```diff
@@ -142,223 +159,209 @@ task cucumberTestReport(type: TestReport) {
...
 wrapper {
-    gradleVersion = '4.10.2'
+    gradleVersion = "6.0.1"
 }
 
 task stage(dependsOn: 'bootWar') {
 }
 
 bootWar.dependsOn war
-compileJava {
```
Comment (v-kyrychenko) [inline build.gradle]: better to leave this section in the same place

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,5 +1,6 @@
+#Tue Dec 10 12:46:04 EET 2019
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.0.1-all.zip
```
Comment (v-kyrychenko) [inline gradle/wrapper/gradle-wrapper.properties]: let's use gradle-6.0.1-bin.zip

File: src/main/java/com/icthh/xm/ms/balance/config/AsyncConfiguration.java [hunk 4]
```diff
@@ -35,7 +36,7 @@ public AsyncConfiguration(JHipsterProperties jhipsterProperties) {
 
     @Override
     @Bean(name = "taskExecutor")
-    public Executor getAsyncExecutor() {
+    public TaskExecutor getAsyncExecutor() {
```
Comment (v-kyrychenko) [inline src/main/java/com/icthh/xm/ms/balance/config/AsyncConfiguration.java]: why this change is made?

File: src/main/java/com/icthh/xm/ms/balance/domain/BalanceChangeEvent.java [hunk 5]
```diff
@@ -40,32 +43,52 @@
...
     private Long id;
 
-    private Long balanceId;
-    private String balanceKey;
-    private String balanceTypeKey;
-    private Long balanceEntityId;
-
-    @Column(precision = 10, scale = 2)
+    @NotNull
+    @Column(name = "amount_delta", precision = 10, scale = 2, nullable = false)
```
Comment (v-kyrychenko) [inline src/main/java/com/icthh/xm/ms/balance/domain/BalanceChangeEvent.java:47]: nullable = false is missing in old version. Why you add this configuration?

File: src/main/java/com/icthh/xm/ms/balance/domain/BalanceChangeEvent.java [hunk 6]
```diff
@@ -40,32 +43,52 @@
...
     private BigDecimal amountDelta;
 
+    @NotNull
     @Enumerated(STRING)
+    @Column(name = "operation_type", nullable = false)
```
Comment (v-kyrychenko) [inline src/main/java/com/icthh/xm/ms/balance/domain/BalanceChangeEvent.java:52]: the same question about nullable

File: src/main/java/com/icthh/xm/ms/balance/domain/BalanceChangeEvent.java [hunk 7]
```diff
@@ -40,32 +43,52 @@
...
+    @Column(name = "amount_before", precision = 10, scale = 2)
     private BigDecimal amountBefore;
 
-    @Column(precision = 10, scale = 2)
+    @Column(name = "amount_after", precision = 10, scale = 2)
     private BigDecimal amountAfter;
 
+    @NotNull
+    @Column(name = "balance_id", nullable = false)
+    private Long balanceId;
```
Comment (v-kyrychenko) [inline src/main/java/com/icthh/xm/ms/balance/domain/BalanceChangeEvent.java:79]: leave this fields group in the same position as in original class

File: src/main/java/com/icthh/xm/ms/balance/service/FilterConverter.java [hunk 8]
```diff
@@ -67,20 +67,20 @@ private static QueryPart toJpql(Map<String, Filter> filterMap) {
                                                Operation.GREATER_THAN,
                                                rangeFilter.getGreaterThan()));
             }
-            if (rangeFilter.getGreaterOrEqualThan() != null) {
+            if (rangeFilter.getGreaterThanOrEqual() != null) {
```
Comment (v-kyrychenko) [inline src/main/java/com/icthh/xm/ms/balance/service/FilterConverter.java:70]: why this change was made?

File: src/main/java/com/icthh/xm/ms/balance/service/mapper/BalanceChangeEventMapper.java [hunk 9]
```diff
@@ -8,13 +8,19 @@
...
     BalanceChangeEventDto toDto(BalanceChangeEvent entity);
 
-    @Override
     @Mapping(target = "metadata.metadata", source = "metadata" )
+    @Mapping(target = "id", ignore = true)
+    @Mapping(target = "pocketChangeEvents", ignore = true)
+    @Mapping(target = "executedByUserKey", ignore = true)
+    @Mapping(target = "balanceTypeKey", ignore = true)
+    @Mapping(target = "balanceKey", ignore = true)
+    @Mapping(target = "balanceEntityId", ignore = true)
```
Comment (v-kyrychenko) [inline src/main/java/com/icthh/xm/ms/balance/service/mapper/BalanceChangeEventMapper.java:23]: please explain why this new annotations is needed

File: src/main/java/com/icthh/xm/ms/balance/service/mapper/BalanceMapper.java [hunk 10]
```diff
@@ -8,10 +8,11 @@
 /**
  * Mapper for the entity Balance and its DTO BalanceDTO.
  */
-@Mapper(componentModel = "spring", uses = {})
+@Mapper(componentModel = "spring")
 public interface BalanceMapper extends EntityMapper<BalanceDTO, Balance> {
 
-
+    @Mapping(target = "removePockets", ignore = true)
+    @Mapping(target = "removeMetrics", ignore = true)
```
Comment (v-kyrychenko) [inline src/main/java/com/icthh/xm/ms/balance/service/mapper/BalanceMapper.java:15]: please explain why this new annotations is needed

File: src/main/java/com/icthh/xm/ms/balance/service/mapper/PocketMapper.java [hunk 11]
```diff
@@ -15,6 +15,7 @@
     @Mapping(source = "metadata.metadata", target = "metadata")
     PocketDTO toDto(Pocket pocket);
 
+    @Mapping(target = "subtractAmount", ignore = true)
```
Comment (v-kyrychenko) [inline src/main/java/com/icthh/xm/ms/balance/service/mapper/PocketMapper.java:18]: please explain why this new annotations is needed

File: src/test/java/com/icthh/xm/ms/balance/web/rest/BalanceResourceIntTest.java [hunk 12]
```diff
@@ -139,8 +139,7 @@ public void BeforeTransaction() {
     }
 
     @After
-    @Override
-    public void finalize() {
+    public void tearDown() {
```
Comment (v-kyrychenko) [inline src/test/java/com/icthh/xm/ms/balance/web/rest/BalanceResourceIntTest.java:142]: why name of the method is changed?

File: src/test/resources/config/application.yml [hunk 13]
```diff
@@ -46,6 +47,8 @@ spring:
             hibernate.cache.use_query_cache: false
             hibernate.generate_statistics: true
             hibernate.hbm2ddl.auto: validate
+            # see org.hibernate.cfg.AvailableSettings.PREFER_GENERATOR_NAME_AS_DEFAULT_SEQUENCE_NAME
+            hibernate.model.generator_name_as_sequence_name: false
```
Comment (v-kyrychenko) [inline src/test/resources/config/application.yml:51]: and what about prod profile? I guess the same changes must be done in application-prod.yml

Please try to save some entity using rest api and you will see result

**Review by v-kyrychenko — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (saylenty) [reply build.gradle]: This library is need for successful compilation: 
eg. `com.vladmihalcea.hibernate.type.json.JsonBinaryType;` in `JsonbTypeRegistrator.java:7`

File: build.gradle (hunk 2 above)
Comment (saylenty) [reply build.gradle]: Ok

File: gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (saylenty) [reply gradle/wrapper/gradle-wrapper.properties]: Ok

File: src/main/java/com/icthh/xm/ms/balance/domain/BalanceChangeEvent.java (hunk 5 above)
Comment (saylenty) [reply src/main/java/com/icthh/xm/ms/balance/domain/BalanceChangeEvent.java:47]: Because it's actually not nullable from liqubase scripts

File: src/main/java/com/icthh/xm/ms/balance/service/mapper/BalanceChangeEventMapper.java (hunk 9 above)
Comment (saylenty) [reply src/main/java/com/icthh/xm/ms/balance/service/mapper/BalanceChangeEventMapper.java:23]: Added this annotations to avoid mapstruct warnings during compilation.
MapStruct says that those field will be ignored during mapping. These annotations doesn't change the behavior of the application, but disable MapStruct mapping warnings.

File: src/main/java/com/icthh/xm/ms/balance/service/mapper/BalanceMapper.java (hunk 10 above)
Comment (saylenty) [reply src/main/java/com/icthh/xm/ms/balance/service/mapper/BalanceMapper.java:15]: Same as above

File: src/main/java/com/icthh/xm/ms/balance/service/mapper/PocketMapper.java (hunk 11 above)
Comment (saylenty) [reply src/main/java/com/icthh/xm/ms/balance/service/mapper/PocketMapper.java:18]: Same as above

File: src/test/java/com/icthh/xm/ms/balance/web/rest/BalanceResourceIntTest.java (hunk 12 above)
Comment (saylenty) [reply src/test/java/com/icthh/xm/ms/balance/web/rest/BalanceResourceIntTest.java:142]: method `finalize` is deprecated since `java 9`. Don't know need to use it in the tests. General practise is to name `setup/tearDown` methods for test inialization and destroy functionality

File: src/main/java/com/icthh/xm/ms/balance/config/AsyncConfiguration.java (hunk 4 above)
Comment (saylenty) [reply src/main/java/com/icthh/xm/ms/balance/config/AsyncConfiguration.java]: taskExecutor bean is injected into: `com.icthh.xm.ms.balance.config.DatabaseConfiguration#liquibase` line 111, param 1.
It doesn't work the way it was before, but does with this change

File: src/test/resources/config/application.yml (hunk 13 above)
Comment (saylenty) [reply src/test/resources/config/application.yml:51]: Indeed. Fixed.

Comment (saylenty) [reply src/test/resources/config/application.yml]: Done

File: src/main/java/com/icthh/xm/ms/balance/service/FilterConverter.java (hunk 8 above)
Comment (saylenty) [reply src/main/java/com/icthh/xm/ms/balance/service/FilterConverter.java:70]: Because these methods were renamed.

**Review by v-kyrychenko — APPROVED** [review]

**Review by amedvedchuk — APPROVED** [review]
